bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlTreeBuilder.java

```diff
@@ -96,8 +96,8 @@ public XmlTreeBuilder(Reader stringReader) {
     }
 
     private void handleXMLStreamException(Exception e) {
-        // todo: do e.getMessage contain all the information? verify
-        throw new BallerinaException(e.getMessage(), e);
+        String reason = e.getCause() == null ? e.getMessage() : e.getCause().getMessage();
```

Comment:
I have seen, the message is also `null` with some exceptions, we need to handle that case as well.

Reply:
Will address this.